File: app/src/main/java/develop/y/zhzl/list/widget/TabFragment.java
package develop.y.zhzl.list.widget;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;

import develop.y.zhzl.R;
import framework.base.BaseFragment;
import framework.mvp.PresenterCompat;

/**
 * by y on 2016/8/7.
 */
public class TabFragment extends BaseFragment {

    private TabLayout tabLayout;
    private ViewPager viewPager;
    private TabAdapter adapter;

    public static TabFragment newInstance(String type) {
        TabFragment imageListTabFragment = new TabFragment();
        Bundle bundle = new Bundle();
        bundle.putString(FRAGMENT_INDEX, type);
        imageListTabFragment.setArguments(bundle);
        return imageListTabFragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle bundle = getArguments();
        if (null != bundle && !bundle.isEmpty()) {
            type = bundle.getString(FRAGMENT_INDEX);
        }
    }

    @Override
    protected PresenterCompat initPresenter() {
        return null;
    }

    @Override
    protected void initById() {
        tabLayout = (TabLayout) getView(R.id.tab_layout);
        viewPager = (ViewPager) getView(R.id.viewPager);
    }

    @Override
    protected void initActivityCreated() {
        adapter = new TabAdapter(getChildFragmentManager(), type);
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_tab;
    }
}

File: app/src/main/java/develop/y/zhzl/list/widget/TabFragment.kt
package develop.y.zhzl.list.widget

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import develop.y.zhzl.R

/**
 * by y on 2016/8/7.
 */
class TabFragment : Fragment() {

    private var tabLayout: TabLayout? = null
    private var viewPager: ViewPager? = null
    private var adapter: TabAdapter? = null

    private var type: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val bundle = arguments
        if (null != bundle && !bundle.isEmpty) {
            type = bundle.getString("index")
        }
    }


    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val inflate = inflater!!.inflate(R.layout.fragment_tab, null)
        tabLayout = inflate.findViewById(R.id.tab_layout) as TabLayout
        viewPager = inflate.findViewById(R.id.viewPager) as ViewPager
        return inflate
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        adapter = TabAdapter(childFragmentManager, type!!)
        viewPager!!.adapter = adapter
        tabLayout!!.setupWithViewPager(viewPager)
    }

    companion object {

        fun newInstance(type: String): TabFragment {
            val imageListTabFragment = TabFragment()
            val bundle = Bundle()
            bundle.putString("index", type)
            imageListTabFragment.arguments = bundle
            return imageListTabFragment
        }
    }

}
